### PR TITLE
ops.define_tensor can take contiguity as single boolean.

### PR DESCRIPTION
**Pull request by wujingyue**
As a follow-up to https://github.com/NVIDIA/Fuser/pull/3116#discussion_r1789980138.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -1005,24 +1017,17 @@ void initNvFuserPythonBindings(PyObject* module) {
           "define_tensor",
           [](FusionDefinition& self,
              const std::vector<int64_t>& shape,
-             std::vector<std::optional<bool>> contiguity = {},
+             std::vector<std::optional<bool>> contiguity,
```
Comment (jjsjann123) [inline csrc/python_frontend/python_bindings.cpp:1020]: 👏

**Review by jjsjann123 — APPROVED** [review]
LGTM, thanks a lot for picking up the API change.

File: tests/python/test_multidevice.py [hunk 2]
```diff
@@ -224,67 +224,67 @@ def definition(self) -> None:
...
         self.layernorm0_bias = self.define_tensor(
             shape=[e],
-            contiguity=[True],
+            contiguity=True,
             dtype=DataType.BFloat16,
         )
         self.mha_linear0_weight = self.define_tensor(
-            shape=[e * 3, e],
-            contiguity=[True, True],
+            shape=[d, e * 3 // d, e],
```
Comment (jjsjann123) [inline tests/python/test_multidevice.py:241]: out of curiosity, what's the reason for not having these sharded earlier? i.e. change in this PR on API isn't expanding any functionality.

Comment (wujingyue) [reply tests/python/test_multidevice.py:241]: That belongs to a different change. I'll rebase. Thanks for catching that!

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

**Pull request merged**